services/core/ActuatorAgent/actuator/agent.py

```diff
@@ -712,33 +716,51 @@ def _update_device_state_and_schedule(self, now):
+        #If we are being called as part of a regularly scheduled publish
+        #we ignore our previous publish schedule time.
+        if device_only is None and publish:
+            self._update_event_time = None
+
         schedule_next_event_time = self._schedule_manager.get_next_event_time(
             now)
         _log.debug("schedule_next_event_time is {}".format(
             schedule_next_event_time))
         new_update_event_time = self._get_ajusted_next_event_time(
             now,
-            schedule_next_event_time)
+            schedule_next_event_time,
+            self._update_event_time)
         _log.debug("new_update_event_time is {}".format(
             new_update_event_time))
-        for device, state in self._device_states.iteritems():
-            _log.debug("device, state -  {}, {}".format(device, state))
-            header = self._get_headers(state.agent_id,
-                                       time=utils.format_timestamp(now),
-                                       task_id=state.task_id)
-            header['window'] = state.time_remaining
-            topic = topics.ACTUATOR_SCHEDULE_ANNOUNCE_RAW.replace('{device}',
-                                                                  device)
-            self.vip.pubsub.publish('pubsub', topic, headers=header)
+
+        if publish:
+            device_states = []
+            if device_only is not None:
+                if device_only in self._device_states:
+                    device_states.append((device_only, self._device_states[device_only]))
+            else:
+                device_states = self._device_states.iteritems()
+
+            for device, state in device_states:
+                _log.debug("device, state -  {}, {}".format(device, state))
+                header = self._get_headers(state.agent_id,
+                                           time=utils.format_timestamp(now),
+                                           task_id=state.task_id)
+                header['window'] = state.time_remaining
+                topic = topics.ACTUATOR_SCHEDULE_ANNOUNCE_RAW.replace('{device}',
+                                                                      device)
+                self.vip.pubsub.publish('pubsub', topic, headers=header)
 
         if self._update_event is not None:
             # This won't hurt anything if we are canceling ourselves.
             self._update_event.cancel()
+        self._update_event_time = new_update_event_time
         self._update_event = self.core.schedule(new_update_event_time,
                                                 self._update_schedule_state,
                                                 new_update_event_time)
 
-    def _get_ajusted_next_event_time(self, now, next_event_time):
+    def _get_ajusted_next_event_time(self, now, next_event_time, previously_scheduled_time):
         _log.debug("_get_adjusted_next_event_time")
```

Comment:
Consider changing so that you spell the word adjusted correctly in the function name.